app/bundles/EmailBundle/Controller/EmailController.php

```diff
@@ -1133,7 +1133,7 @@ public function builderAction(Request $request, SlotsHelper $slotsHelper, $objec
 
         $this->processSlots($slotsHelper, $slots, $entity);
 
-        $logicalName = $this->factory->getHelper('theme')->checkForTwigTemplate('@themes/'.$template.'/html/email.html.twig');
+        $logicalName = $this->factory->getHelper('theme')->checkForTwigTemplate('@themes/'.$template.'/html/email.mjml.twig');
```

Comment:
I think it should be something like:
```suggestion
        if ($this->factory->getHelper('theme')->exists('@themes/'.$template.'/html/email.mjml.twig')) {
            $logicalName = $this->factory->getHelper('theme')->checkForTwigTemplate('@themes/'.$template.'/html/email.mjml.twig');
        } else {
            $logicalName = $this->factory->getHelper('theme')->checkForTwigTemplate('@themes/'.$template.'/html/email.html.twig');
        }
```
I can see that some themes have the email.html.twig with the same content as theme.mjml.twig probably because of the issue reported by the failing test. I don't actually see the `email.mjml.twig` being used anywhere in the code base. Or perhaps we could just store the MJML code inside the theme.html.twig file and be done with it.

Reply:
Maybe I understood the deprecation comment wrongly (GrapesJSController.php)

![image](https://github.com/user-attachments/assets/ffcf0e0a-7fcf-4008-bb47-e215c7a32784)

![image](https://github.com/user-attachments/assets/d231cdfa-2131-45aa-a609-786d7569ee6a)

Is it about changing themes to email.mjml.twig or just placing MJML content within the html.twig file?

Reply:
Good catch! I was looking for such code but couldn't find it. But this change is the BC break so it would have to land in Mautic 6.0. Is that OK?

Reply:
From what you said, I understand that I have to revert these changes to PHP files and rename the blank theme again to .html.twig

is it right?

Reply:
I don't know what the official way to solve it is. I'd go with what the other themes do and create a duplicate html.twig file with the same content in mjml.twig file. We can remove all those in M6.

Reply:
just reverted the commits, later we can rename all files
i really need to move some improvements forward, so these details will still pending

Reply:
@escopecz also thanks for having a look :)